python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -133,22 +147,36 @@ def runtimeEvalString(this_obj, attr_name, val):
 
 
 def raw_eval(val):
+    ## Attempts to evaluate the val object and return the object it evaluates to if it is a Python object
+    ## Makes use of basic caching as we don't expect that things at this level should change.
+    from copy import deepcopy
+    if val in _eval_cache:
+        return deepcopy(_eval_cache[val])
+
     try:
+        ## Catch common strings which shouldn't be evaluated anyway and return them asif an exception occured
+        if val in ['', '.', 'new']:
+            _eval_cache[val] = val
+            return deepcopy(val)
```

Comment:
If the eval is evaluating '' it throws an exception and that's taken to be a string which can't be eval-led such as 'hello' so the returned value is the string in this case from this function, even if it did as it was having to deal with the likes of 'hello' until now it's difficult to know to distinguish between this and the special case of '' without a check